packages/lib-classifier/src/components/Classifier/components/SubjectViewer/SubjectViewer.spec.js

```diff
@@ -1,60 +1,52 @@
 import asyncStates from '@zooniverse/async-states'
-import { shallow } from 'enzyme'
-
 import { SubjectViewer } from './SubjectViewer'
-import SingleImageViewer from './components/SingleImageViewer'
-import JSONDataViewer from './components/JSONDataViewer'
 
 describe('Component > SubjectViewer', function () {
   it('should render without crashing', function () {
-    const wrapper = shallow(<SubjectViewer />)
-    expect(wrapper).to.be.ok()
+    render(<SubjectViewer />)
+    expect(screen).to.be.ok()
   })
 
   it('should render nothing if the subject store is initialized', function () {
-    const wrapper = shallow(<SubjectViewer subjectQueueState={asyncStates.initialized} />)
-    expect(wrapper.type()).to.be.null()
+    const { container } = render(<SubjectViewer subjectQueueState={asyncStates.initialized} />)
+    expect(container.firstChild).to.be.null()
   })
-
+  
   it('should render a loading indicator if the subject store is loading', function () {
-    const wrapper = shallow(<SubjectViewer subjectQueueState={asyncStates.loading} />)
-    expect(wrapper.text()).to.equal('SubjectViewer.loading')
+    render(<SubjectViewer subjectQueueState={asyncStates.loading} />)
+    expect(screen.getByText('SubjectViewer.loading')).to.exist()
   })
 
   it('should render nothing if the subject store errors', function () {
-    const wrapper = shallow(<SubjectViewer subjectQueueState={asyncStates.error} />)
-    expect(wrapper.type()).to.be.null()
+    const { container } = render(<SubjectViewer subjectQueueState={asyncStates.error} />)
+    expect(container.firstChild).to.be.null()
   })
 
-  it('should render a subject viewer if the subject store successfully loads', function () {
-    const wrapper = shallow(<SubjectViewer subjectQueueState={asyncStates.success} subject={{ viewer: 'singleImage' }} />)
-    expect(wrapper.find(SingleImageViewer)).to.have.lengthOf(1)
-  })
-
-  it('should pass along the viewer configuration', function () {
-    const viewerConfiguration = {
-      zoomConfiguration: {
-        direction: 'both',
-        minZoom: 1,
-        maxZoom: 10,
-        zoomInValue: 1.2,
-        zoomOutValue: 0.8
-      }
-    }
-
-    const wrapper = shallow(<SubjectViewer subjectQueueState={asyncStates.success} subject={{ viewer: 'variableStar', viewerConfiguration }} />)
-    expect(wrapper.find(JSONDataViewer).props().viewerConfiguration).to.deep.equal(viewerConfiguration)
+  it('should render the VolumetricViewer if the subject store successfully loads', function () {
+    render(<SubjectViewer
+      subjectQueueState={asyncStates.success}
+      isVolumetricViewer={true}
+      subject={{
+        id: 'mock-id',
+        locations: [
+          { 'image/jpeg': `https://www.zooniverse.org/mock-subjects/file-mock-id.jpg` }
+        ]
+      }}
+    />)
+    expect(screen.getByText('Suspense boundary')).to.exist()
+    expect(screen.findByTestId('subject-viewer-volumetric')).to.exist()
```

Comment:
The expectation on line 38 is a false positive. I can change that line to `findByTestId('subject-viewer-volumetriccccc').to.exist()` and it will still be true. See the documented Issue about `findBy` queries: https://github.com/zooniverse/front-end-monorepo/issues/6383

This unit test needs reworking before this PR can be merged.

Reply:
Great catch! Fixed with adding `async` in the spec and a proper return element that doesn't dive into WebGL/Canvas context rendering in the testrunner.